features/step_definitions/files_steps.rb

```diff
@@ -30,3 +30,10 @@
   expect(uncommitted_files).to eql [file_name]
   expect(IO.read file_name).to eql content
 end
+
+
+Then(/^my uncommitted file "(.+?)" is still stashed away$/) do |file_name|
+  expect(uncommitted_files).to_not include file_name
```

Comment:
I don't want to hold this shotgun refactor up any longer, since I just had to resolve a few dozen merge conflicts while updating the branch. I have extracted this (great) idea into #488, and will implement it right after this PR goes in in a separate feature branch.